packages/prerender-fargate/lib/prerender/server.js

```diff
@@ -36,8 +36,114 @@ server.use({
 });
 
 server.use(prerender.blacklist());
-server.use(prerender.httpHeaders());
-server.use(prerender.removeScriptTags());
-server.use(s3Cache);
+
+if (process.env.ENABLE_REDIRECT_CACHE.toLowerCase() === 'true'){
+    server.use(prerender.removeScriptTags());
+    var he = require('he');
+    var s3 = new (require('aws-sdk')).S3({params:{Bucket: process.env.S3_BUCKET_NAME}});
+    server.use({
+            requestReceived: function(req, res, next) {
+                if(req.method !== 'GET' && req.method !== 'HEAD') {
+                    console.log("skipping requestReceived from S3 Cache... ")
+                    return next();
+                }
+
+                var key = req.prerender.url;
+
+                if (process.env.S3_PREFIX_KEY) {
+                    key = process.env.S3_PREFIX_KEY + '/' + key;
+                }
+
+                s3.getObject({
+                        Key: key
+                }, function (err, result) {
+
+                    if (!err && result) {
+                        console.log(result.Metadata);
+                        console.log("Found cached object: " + key);
+                        if (result.Metadata.location){
+                            res.setHeader('Location', result.Metadata.location);
+                        }
+                        return res.send(result.Metadata.httpreturncode, result.Body);
+                    } else {
+                        console.error(err);
+                    }
+
+                    next();
+                });
+            },
+
+            pageLoaded: function(req, res, next) {
+                const statusCodesToCache = ['200', '301', '302'];
+                let metaTagStatusCode = 200;
+                let location = '';
+
+                if (req.prerender.content && req.prerender.renderType == 'html') {
+                    let statusMatch = /<meta[^<>]*(?:name=['"]prerender-status-code['"][^<>]*content=['"]([0-9]{3})['"]|content=['"]([0-9]{3})['"][^<>]*name=['"]prerender-status-code['"])[^<>]*>/i,
+                    headerMatch = /<meta[^<>]*(?:name=['"]prerender-header['"][^<>]*content=['"]([^'"]*?): ?([^'"]*?)['"]|content=['"]([^'"]*?): ?([^'"]*?)['"][^<>]*name=['"]prerender-header['"])[^<>]*>/gi,
+                    head = req.prerender.content.toString().split('</head>', 1).pop(),
+                    // statusCode = 200,
+                    match;
```

Comment:
Just for clarity I'd put a `let` on each line rather than using the comma syntax. I can be a bit hard to read especially with the long regexes :sweat_smile:

Reply:
Also if a variable isn't going to change you should use the `const` keyword rather than `let`